src/transport/GroupPeerMessageCounter.cpp

```diff
@@ -172,6 +172,29 @@ CHIP_ERROR GroupPeerTable::RemovePeer(FabricIndex fabricIndex, NodeId nodeId, bo
     return err;
 }
 
+CHIP_ERROR GroupPeerTable::RemoveFabricIndex(FabricIndex fabricIndex)
+{
+    CHIP_ERROR err = CHIP_ERROR_NOT_FOUND;
+
+    if (fabricIndex == kUndefinedFabricIndex)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    for (uint32_t it = 0; it < CHIP_CONFIG_MAX_FABRICS; it++)
+    {
+        if (fabricIndex == mGroupFabrics[it].mFabricIndex)
+        {
+            new (&mGroupFabrics[it]) GroupFabric();
```

Comment:
But ... doesn't this leave a hole in the array, of exactly the sort that `GroupPeerTable::RemovePeer` tries to avoid?  Seems like this should share logic with that method....

Reply:
Well it's a destructor with an array of delegate. The logic was that if the delegate is not owned by the FabricTable it could be owned by almost anything  so there is no point in preserving the linked list if two delegates are owned by different objects. Like why would the delegate from the server keep a reference to the delegate from a cluster implementation? Also should you delete the Fabric table one could assume that the linked list stored inside the destroyed FabricTable isn't valid anymore.